GiasPostcodeSearch/GiasPostcodeSearchService.cs

```diff
@@ -8,35 +13,37 @@ public class GiasPostcodeSearchService : IHostedService
 {
     private readonly IHostApplicationLifetime _host;
     private readonly ILogger<GiasPostcodeSearchService> _logger;
+    private readonly IServiceScopeFactory _scopeFactory;
     private readonly HttpClient _httpClient;
-    private readonly ISchoolDataProvider _schoolDataProvider;
 
     public GiasPostcodeSearchService(
         IHostApplicationLifetime host,
         ILogger<GiasPostcodeSearchService> logger,
-        HttpClient httpClient,
-        ISchoolDataProvider schoolDataProvider)
+        IServiceScopeFactory scopeFactory,
+        HttpClient httpClient)
     {
         _host = host;
         _logger = logger;
+        _scopeFactory = scopeFactory;
         _httpClient = httpClient;
-        _schoolDataProvider = schoolDataProvider;
     }
 
     public async Task StartAsync(CancellationToken cancellationToken)
     {
-        var schoolData = (await _schoolDataProvider.GetSchoolDataAsync(cancellationToken)).Where(x =>
-            new[]
-            {
-                PhaseOfEducation.Primary, PhaseOfEducation.MiddleDeemedPrimary, PhaseOfEducation.Secondary,
-                PhaseOfEducation.MiddleDeemedSecondary, PhaseOfEducation.AllThrough
-            }.Contains(x.PhaseOfEducation)).ToArray();
+        using var scope = _scopeFactory.CreateScope();
+        var dbContext = scope.ServiceProvider.GetRequiredService<NtpDbContext>();
+
+        var schools = await dbContext.Schools.Where(e => new[]
```

Comment:
Possibly worth extracting the array into a local variable.  It won't impact performance much but might make it slightly easier to read what the `Where` is doing.

Reply:
Good idea. Done